prov/rxm/src/rxm_ep.c

```diff
@@ -1763,6 +1763,16 @@ static int rxm_ep_bind(struct fid *ep_fid, struct fid *bfid, uint64_t flags)
 		ret = ofi_ep_bind_av(&rxm_ep->util_ep, av);
 		if (ret)
 			return ret;
+		ret = fi_listen(rxm_ep->msg_pep);
+		if (ret) {
+			FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+				"Unable to set msg PEP to listen state\n");
+			return ret;
+		}
+
+		rxm_ep->util_ep.cmap = rxm_conn_cmap_alloc(rxm_ep);
```

Comment:
Does is matter that we start listening before the cmap has been allocated?

Reply:
Nice point. We should ensure that no connections events are received until EQ isn't polled

Reply:
Sorry, I thought I had replied to this comment yesterday. We need listen to precede it as cmap alloc calls fi_getname on the passive endpoint. The endpoint needs to be in the listen state for it to have a valid name.

Reply:
Sorry, I missed this. There is a comment in the code regarding this

I prepared #4254 to address this issue